### DISCUSSION TITLE
Not able to establish connection between  Border Router and nRF5340 DK end device

### DISCUSSION THREAD

**Discussion by vipindas1984** [openthread/openthread#8430] [Q&A] [answered]
Hi all,

I am trying to build a very simple thread network using nRF5340 DK. For that I setup the OTBR using raspberry pi 4 and nRF52840 dongle. I followed the steps mentioned in the following pages: [https://openthread.io/codelabs/openthread-border-router#1](url) && [https://openthread.io/guides/border-router/build](url) 

I used the Raspberry Pi Desktop(OS) and I flashed the ot-rcp.hex for RCP device. I could see the RCP state as 'leader'. Also I was able to form a Thread Network using OTBR Web GUI as per the page: [https://openthread.io/codelabs/openthread-border-router#2](url)

In order to configure the Thread end device I built and flashed **Thread:CLI** sample on my nRF5340 DK with USB as transport. But I am not able to connect to Thread end device to configure the Thread Network parameters. Can someone share me any useful tips/links to properly connect to end device configured with USB transport. 

Thanks in advance.

Vipin Das

**Comment by jwhui** [comment, marked as answer]
Given that this is specific to the nRF5340 DK, I would post this question in [Nordic DevZone](https://devzone.nordicsemi.com/).

**Reply by vipindas1984** [reply]
Thank you Jonathan. But I have a specific question wrt OpenThread. Is it possible to setup a small OpenThread Network with the support of  one OTBR(RPi & nRF52840 Dongle) and One Thread end device(nRF5340DK). If so which example and what configuration I should consider.

**Reply by jwhui** [reply]
The [OpenThread Border Router codelab](https://openthread.io/codelabs/openthread-border-router) describes how to setup a small network with one OTBR and one Thread device.

**Reply by vipindas1984** [reply]
Hi Jonathan,

Thanks again. Not sure whether my question is specific to this discussion forum. But let me ask my doubts. 

I followed that page and flashed the Thread:CLI sample by enabling -DOT_SRP_CLIENT=ON -DOT_ECDSA=ON on my end device here it is my nRF5340 DK. But I was not able to set "dataset set active" on my SRP Client End Device because I don't see the "SRP client node" as mentioned in the page. 

For me I see the this. 
uart:~$ dataset set active 000300000b35060004001fffe00208dead0beef00cafe0708fddead00beef000005100011223344566778899aabbccddeeff03096f745f7a6570687972002abcd04105e6973080b858c2638f4227f67b8d7980c0402a0f7f80e080000000000000000
dataset: command not found

Any clue on this. Did I miss any important steps. 

Thanks 

Vipin Das

**Reply by jwhui** [reply]
If you are using the Nordic SDK, you can try prefixing all OT CLI commands with `ot`. See the [nRF Connect SDK CLI documentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/openthread/cli/README.html) for examples.

**Reply by vipindas1984** [reply]
Hi Jonathan,

Thank you so much. Your points helped. I mixed up the OpenThread documentation and NRF Connect SDK documentation. Now things are getting clear. 

Regards

Vipin Das
